pkg/trustedresources/verify.go

```diff
@@ -169,12 +173,24 @@ func getMatchedPolicies(resourceName string, source string, policies []*v1alpha1
 //  1. If multiple policies match, the resource must satisfy all the "enforce" policies to pass verification. The matching "enforce" policies are evaluated using AND logic.
 //     Alternatively, if the resource only matches policies in "warn" mode, it will still pass verification and only log a warning if these policies are not satisfied.
 //  2. To pass one policy, the resource can pass any public keys in the policy. We use OR logic on public keys of one policy.
-func verifyResource(ctx context.Context, resource metav1.Object, k8s kubernetes.Interface, signature []byte, matchedPolicies []*v1alpha1.VerificationPolicy) error {
+func verifyResource(ctx context.Context, resource metav1.Object, k8s kubernetes.Interface, signature []byte, matchedPolicies []*v1alpha1.VerificationPolicy) *VerificationResult {
+	logger := logging.FromContext(ctx)
+	var warnPolicies []*v1alpha1.VerificationPolicy
+	var enforcePolicies []*v1alpha1.VerificationPolicy
 	for _, p := range matchedPolicies {
+		if p.Spec.Mode == v1alpha1.ModeWarn {
+			warnPolicies = append(warnPolicies, p)
+		} else {
+			enforcePolicies = append(enforcePolicies, p)
+		}
+	}
+
+	// first evaluate all enforce policies. Return VerificationError type of VerificationResult if any policy fails.
+	for _, p := range enforcePolicies {
 		passVerification := false
 		verifiers, err := verifier.FromPolicy(ctx, k8s, p)
 		if err != nil {
-			return fmt.Errorf("failed to get verifiers from policy: %w", err)
+			return &VerificationResult{VerificationResultType: VerificationError, Err: fmt.Errorf("failed to get verifiers from policy: %w", err)}
```

Comment:
would it make sense to return a multierror of some kind, instead of the first error?

Reply:
I was also thinking about this yesterday and I think it makes sense to return all failed policies in the error, 
maybe add a todo this could be a future work to ?

Reply:
sounds good! can you create an issue to track this and link here in the comments?

Reply:
added a todo